### ISSUE TITLE
Breaking change in System.Text.Json.dll,  infinite recursion in System.Text.Json.Serialization.JsonConverter. Works fine in .NET 6

### ISSUE THREAD

**Issue by chabiss** [dotnet/runtime#75673]
### Description
Breaking change in System.Text.Json.JsonSerializerOptions causing stack overflow.  Work fine in .NET 6
```
>	JsonSerializationIssue.dll!JsonSerializationIssue.MessageJsonConverter.Write(System.Text.Json.Utf8JsonWriter writer, JsonSerializationIssue.Message value, System.Text.Json.JsonSerializerOptions options) Line 74	C#
 	System.Text.Json.dll!System.Text.Json.Serialization.JsonConverter<object>.TryWrite(System.Text.Json.Utf8JsonWriter writer, object value, System.Text.Json.JsonSerializerOptions options, ref System.Text.Json.WriteStack state)	Unknown
 	System.Text.Json.dll!System.Text.Json.Serialization.JsonConverter<System.__Canon>.WriteCore(System.Text.Json.Utf8JsonWriter writer, System.__Canon value, System.Text.Json.JsonSerializerOptions options, ref System.Text.Json.WriteStack state)	Unknown
 	System.Text.Json.dll!System.Text.Json.JsonSerializer.WriteCore<object>(System.Text.Json.Utf8JsonWriter writer, object value, System.Text.Json.Serialization.Metadata.JsonTypeInfo<object> jsonTypeInfo)	Unknown
 	System.Text.Json.dll!System.Text.Json.JsonSerializer.Serialize<object>(System.Text.Json.Utf8JsonWriter writer, object value, System.Text.Json.JsonSerializerOptions options)	Unknown
 	JsonSerializationIssue.dll!JsonSerializationIssue.MessageJsonConverter.Write(System.Text.Json.Utf8JsonWriter writer, JsonSerializationIssue.Message value, System.Text.Json.JsonSerializerOptions options) Line 74	C#
```
### Configuration
Open the attached project and run it in VS [JsonSerializationIssue.zip](https://github.com/dotnet/core/files/9562441/JsonSerializationIssue.zip)

### Regression?

Yes, this is a regression from .NET 6.  The same code runs fine on .NET 6

### Other information

This breaking had affected XAML Hot Reload for MAUI when validating .NET 7
https://github.com/xamarin/UITools/pull/1099

The workaround for this bug is to change the method signature from using 

`JsonSerializer.Serialize (writer, (object)value, options)`

to using 

`JsonSerializer.Serialize (writer, value, value.GetType(), options);`

**Comment by eiriktsarpalis** [comment]
TL;DR I don't believe the stack overflow in question is inherent to the breaking change. As such I don't think we should be documenting it.
